src/main/java/org/broadinstitute/consent/http/service/DatasetService.java

```diff
@@ -57,6 +58,36 @@ public DataSet getDatasetWithPropertiesById(Integer datasetId) {
         return dataset;
     }
 
+    public Optional<DataSet> updateDataset(DataSetDTO dataset, Integer datasetId, Integer userId) {
+        Timestamp now = new Timestamp(new Date().getTime());
+
+        DataSet old = getDatasetWithPropertiesById(datasetId);
+        Set<DataSetProperty> oldProperties = old.getProperties();
+
+        List<DataSetPropertyDTO> dtos = dataset.getProperties();
```

Comment:
Naming nit (naming is always hard): I think this should reflect more about what it is, like `updateDatasetPropertyDTOs` so that you can quickly distinguish that these are on the new dataset.